qemu/tests/balloon_check.py

```diff
@@ -355,6 +356,94 @@ def get_win_mon_free_mem(self, session):
+        :param win_drive_letter: virtio-win disk volume letter.
+        :return: cmd execution status and output
+        """
+        try:
+            status_cmd = self.params["status_balloon_service"] % win_drive_letter
+            logging.debug("Check balloon service status.")
+            status, output = session.cmd_status_output(status_cmd)
+            return (status, output)
+        except ShellCmdError:
+            self.test.error("balloon service status check failed!")
+
+    def install_balloon_service(self, session, win_drive_letter):
+        """
+        Check/Install/Start balloon service for windows guest.
+
+        :param session: shell Object
+        :param win_drive_letter: virtio-win disk volume letter.
+        :return: cmd execution status and output
+        """
+        logging.debug("Check Balloon Service status in guest.")
+        status, output = self.check_balloon_service(session, win_drive_letter)
+        if re.search(r"running", output.lower(), re.M):
+            logging.debug("Balloon service is already running !")
+        elif re.search(r"stop", output.lower(), re.M):
+            logging.debug("Start Balloon Service in guest.")
+            try:
+                self.run_balloon_service(session, win_drive_letter)
+            except ShellCmdError:
+                raise exceptions.TestError("Start balloon service failed!")
+        else:
+            logging.debug("Install Balloon Service in guest.")
+            try:
+                install_cmd = self.params["install_balloon_service"] % win_drive_letter
+                status, output = session.cmd_status_output(install_cmd)
+                return (status, output)
+            except ShellCmdError:
+                self.test.error("Install balloon service failed !")
+
+    def uninstall_balloon_service(self, session, win_drive_letter):
+        """
+        Uninstall balloon service in windows guest
+
+        :param session: shell Object
+        :param win_drive_letter: virtio-win disk volume letter.
+        :return: cmd execution status and output
+        """
+        try:
+            uninstall_cmd = self.params["uninstall_balloon_service"] % win_drive_letter
+            status, output = session.cmd_status_output(uninstall_cmd)
+            return (status, output)
+        except ShellCmdError:
+            self.test.error("Uninstall balloon service failed!")
+
+    def stop_balloon_service(self, session, win_drive_letter):
+        """
+        Stop balloon service in windows guest
+
+        :param session: shell Object
+        :param win_drive_letter: virtio-win disk volume letter.
+        :return: cmd execution status and output
+        """
+        try:
+            stop_cmd = self.params["stop_balloon_service"] % win_drive_letter
+            status, output = session.cmd_status_output(stop_cmd)
+            return (status, output)
+        except ShellCmdError:
+            self.test.error("Stop balloon service failed!")
+
```

Comment:
stop_balloon_service() and run_balloon_service() can be merged to one function.

Reply:
I'd rather separate them and keep them simple so that devs do not need to pass another param to specify which action they want.